hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/ExponentiallyGrowingIdleTimeCalculatorTest.groovy

```diff
@@ -0,0 +1,49 @@
+package pl.allegro.tech.hermes.consumers.consumer
+
+import pl.allegro.tech.hermes.consumers.consumer.idleTime.ExponentiallyGrowingIdleTimeCalculator
+import pl.allegro.tech.hermes.consumers.consumer.idleTime.IdleTimeCalculator
+import spock.lang.Specification
+
+
+class ExponentiallyGrowingIdleTimeCalculatorTest extends Specification {
+
+    def initialIdleTime = 10
+    def maxIdleTime = 1000
+    def base = 2
+
+    IdleTimeCalculator calculator
+
+    def setup() {
+        this.calculator = new ExponentiallyGrowingIdleTimeCalculator(base, initialIdleTime, maxIdleTime)
+    }
+
+    def "initial-idle-time is returned first"() {
+        expect:
+        calculator.idleTime == initialIdleTime
+    }
+
+    def "each consecutive idle-time is multiplied by base"() {
+        expect:
+        (0..6).each {
```

Comment:
you can also write `6.times {`

Reply:
Not really, `(0..6).each` is the same as `7.times`.I'm not sure if that's more readable cause I wanted to express that I'm testing numbers from 0 to 6 and not trying to do something exactly seven times.